Sources/CacheAdvance/CacheReader.swift

```diff
@@ -44,29 +41,29 @@ final class CacheReader {
         reader.offsetInFile
     }
 
-    /// Returns the next encodable message, seeking to the beginning of the next message.
-    func nextEncodedMessage() throws -> Data? {
-        let startingOffset = offsetInFile
-
-        guard startingOffset != offsetInFileAtEndOfNewestMessage else {
-            // We're at the last message.
-            return nil
+    /// Returns the encodable messages in a range
+    ///
+    /// - Parameter startOffset: the start offset to start reading
+    /// - Parameter endOffset: the end offset to stop reading, the default value is EOF if the value is nil.
```

Comment:
another nitty suggestion:
```suggestion
    /// - Parameter endOffset: the offset at which to stop reading. If `nil`, the end offset will be the EOF
```